feat(docs): update micro-frontends deployment guide, versioning, tags, and more (#12)
ZephyrCloudIO/zephyr-documentation: docs/how-to/mf-guide.mdx

import { Steps, Badge } from '@theme';
import { Separator } from "../../components/ui/separator"

# Micro-Frontends with Zephyr - The Complete Guide

> Micro-Frontends' configuration and it's deployment processes are known to be complicated and painful to debug. Zephyr came in not only to help with resolving dynamic URL and development process for applications applying Micro-Frontends architecture, but also deployment, versioning and more. This section will walk through what are the needed configuration to ensure you have a smooth development workflow and deployment process. 

Reviewer: ```suggestion
Micro-Frontends' configuration and deployment processes are known to be complicated and painful to debug. Zephyr Cloud not only to helps with resolving dynamic URL and development process for applications applying Micro-Frontends architecture, but also deployment, versioning and more. This section will walk through the necessary configuration to ensure you have a smooth development workflow and deployment process.
```

Author: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/369b71f277deffafcfd603fa2225db71c8388dc2

## Checklist

<Steps>
### Micro-Frontends Configuration 

> Our first class supported Micro-Frontends architecture is [Module Federation](https://module-federation.io), while other Micro-Frontends architecture might be supported. 

Reviewer: ```suggestion
We have first class support for a Micro-Frontends architecture based on [Module Federation](https://module-federation.io). Other Micro-Frontends architectures may already work or will be supported in the future. 
```

Author: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/cc9fc3d376ebabe752e233c675d250de8b067da1

      name: 'app_01',
      filename: 'remoteEntry.js',
      remotes: {
        app_02: 'app_02@http://localhost:3002/remoteEntry.js',
        app_03: 'app_03@http://localhost:3003/remoteEntry.js',
        app_04: 'app_04@http://localhost:3004/remoteEntry.js',
        app_05: 'app_05@http://localhost:3005/remoteEntry.js',
      },
      exposes: {
        './SideNav': './src/SideNav',
        './Page': './src/Page',
      },
      shared: {
        ...deps,
        '@material-ui/core': {
          singleton: true,
        },
        'react-router-dom': {
          singleton: true,
        },
        'react-dom': {
          singleton: true,
        },
        react: {
          singleton: true,
        },
      },
    }),
```

### Build remotes first 

At the moment. when we are resolving your remote dependencies, we will need the [remotes](/learning/concepts/terminologies#remote-application) to be built first. Be wary that the first built remotes should not be consuming any other remotes. 

In the example below where you have one shell and one remote, you will need to build the remote first. 

![mf_1_circular_dependencies](/mf_guide_1_circular_dependency.png)

> Let's take a look at another example. You have a shell applicaiton, within the shell app you have two modules: **HeaderNav** and **TaskContent**. 

Reviewer: ```suggestion
Let's take a look at another example. You have a shell application, within the shell app you have two modules: **HeaderNav** and **TaskContent**. 
```

Author: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/cc9fc3d376ebabe752e233c675d250de8b067da1

![mf_2_circular_dependencies](/mf_guide_2_circular_dependency.png)

The **HeaderNav** exposes **Logo**, **UserProfileInfo** and **LeftNav**. 

The **TaskContent** exposes **SearchBar**. 

In this scenario, **LeftNav** is consumed by **TaskContent**; **SearchBar** is consumed by **HeaderNav**. 

> You will need to build either the **TaskContent** or the **HeaderNav** first, but comment out the remotes that have yet build and add them back later. 

Reviewer: ```suggestion
You will need to build either the **TaskContent** or the **HeaderNav** first, comment out the remotes that have yet built and add them back later. 
```

Author: https://github.com/ZephyrCloudIO/zephyr-documentation/pull/12/commits/a1627a35c5b5c8212c29bf29937c49730c71ba0a
